afs/afs-ext-base/src/main/java/com/powsybl/afs/ext/base/AbstractScript.java

```diff
@@ -50,6 +52,36 @@ private void processEvents(List<NodeEvent> events, String nodeId, List<ScriptLis
         }
     }
 
+    public void addGenericScript(GenericScript genericScript) {
+        orderedDependencyManager.appendDependencies(INCLUDED_SCRIPTS_DEPENDENCY_NAME, Collections.singletonList(genericScript));
+    }
+
+    public void addScript(T includeScript) {
+        orderedDependencyManager.appendDependencies(INCLUDED_SCRIPTS_DEPENDENCY_NAME, Collections.singletonList(includeScript));
+    }
+
+    public void removeScript(String scriptNodeId) {
+        orderedDependencyManager.removeDependencies(INCLUDED_SCRIPTS_DEPENDENCY_NAME, Collections.singletonList(scriptNodeId));
+    }
+
+    @Override
+    public String readScript(boolean withIncludes) {
+        String ownContent = readScript();
+        if (withIncludes) {
+            String includesScript = orderedDependencyManager
+                    .getDependencies(INCLUDED_SCRIPTS_DEPENDENCY_NAME, AbstractScript.class)
+                    .stream()
+                    // TODO(mathbagu) add commented header for script ? (script project node name ?)
+                    .map(script -> script.readScript(true))
+                    .collect(Collectors.joining(DEFAULT_SCRIPTS_DELIMITER));
+            if (StringUtils.isNotBlank(includesScript)) {
+                includesScript += DEFAULT_SCRIPTS_DELIMITER;
+            }
+            return includesScript + ownContent;
```

Comment:
Do we need a lineSeparator between included scripts and content?

Reply:
yes, if the last included script doesn't end with a newline, it may break things in the overall resulting script..